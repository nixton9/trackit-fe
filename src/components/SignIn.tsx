import React, { useState, useRef } from 'react'
import { SignInSignUpHeader } from './misc/SignInSignUpHeader'
import { Styled } from '../styles/SignInSignUp.styles'
import { ReactComponent as EyeIcon } from '../assets/icons/eye.svg'

const SignIn: React.FC = () => {
  const [emailVal, setEmailVal] = useState('')
  const [passwordVal, setPasswordVal] = useState('')
  const passwordEl = useRef<HTMLInputElement>(null)
  const iconEl = useRef<SVGSVGElement>(null)

  const togglePasswordVisibility = () => {
    const input = passwordEl.current
    const icon = iconEl.current
    if (input && input.type === 'password') {
      input.type = 'text'
      icon?.classList.add('selected')
    } else if (input) {
      input.type = 'password'
      icon?.classList.remove('selected')
    }
  }

  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        <Styled.SignInSignUpText>
          Welcome back, <br />
          <span>sign in to continue</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm>
          <Styled.SignInSignUpForm__Input>
            <input
              type="text"
              placeholder="Email address"
              value={emailVal}
              onChange={e => setEmailVal(e.target.value)}
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <div className="input-wrapper">
              <input
                type="password"
                placeholder="Password"
                value={passwordVal}
                onChange={e => setPasswordVal(e.target.value)}
                ref={passwordEl}
              />
              <EyeIcon onClick={togglePasswordVisibility} ref={iconEl} />
            </div>
            <p>Forgot password?</p>
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Login
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default SignIn
