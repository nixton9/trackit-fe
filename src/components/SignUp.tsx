import React, { useState, useRef } from 'react'
import { SignInSignUpHeader } from './misc/SignInSignUpHeader'
import { Styled } from '../styles/SignInSignUp.styles'
import { ReactComponent as EyeIcon } from '../assets/icons/eye.svg'

const SignUp: React.FC = () => {
  const [nameVal, setNameVal] = useState('')
  const [emailVal, setEmailVal] = useState('')
  const [passwordVal, setPasswordVal] = useState('')
  const passwordEl = useRef<HTMLInputElement>(null)
  const iconEl = useRef<SVGSVGElement>(null)

  const togglePasswordVisibility = () => {
    const input = passwordEl.current
    const icon = iconEl.current
    if (input && input.type === 'password') {
      input.type = 'text'
      icon?.classList.add('selected')
    } else if (input) {
      input.type = 'password'
      icon?.classList.remove('selected')
    }
  }

  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        <Styled.SignInSignUpText>
          Hey there, <br />
          <span>sign up to continue</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm>
          <Styled.SignInSignUpForm__Input>
            <input
              type="text"
              placeholder="Name"
              value={nameVal}
              onChange={e => setNameVal(e.target.value)}
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <input
              type="text"
              placeholder="Email address"
              value={emailVal}
              onChange={e => setEmailVal(e.target.value)}
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <div className="input-wrapper">
              <input
                type="password"
                placeholder="Password"
                value={passwordVal}
                onChange={e => setPasswordVal(e.target.value)}
                ref={passwordEl}
              />
              <EyeIcon onClick={togglePasswordVisibility} ref={iconEl} />
            </div>
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Sign Up
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default SignUp
