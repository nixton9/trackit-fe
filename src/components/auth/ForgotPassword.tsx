import React, { useState } from 'react'
import { SignInSignUpHeader } from '../misc/SignInSignUpHeader'
import { LoadingSpinner } from '../misc/LoadingSpinner'
import { Styled } from '../../styles/SignInSignUp.styles'
import { ReactComponent as ErrorIcon } from '../../assets/icons/error.svg'
import { ReactComponent as CheckIcon } from '../../assets/icons/check.svg'
import { gql, useMutation } from '@apollo/client'

const SUBMIT_PASSWORD_RESET = gql`
  mutation SubmitPasswordReset($email: String!) {
    submitPasswordReset(email: $email) {
      resetToken
    }
  }
`

type SubmitPwData = {
  submitPasswordReset: {
    resetToken: string
  }
}

const ForgotPassword: React.FC = () => {
  const [emailVal, setEmailVal] = useState('')
  const [message, setMessage] = useState('')

  const [submitPwReset, { error, loading }] = useMutation<SubmitPwData>(
    SUBMIT_PASSWORD_RESET,
    {
      variables: { email: emailVal }
    }
  )

  const handleSubmit = (e: React.FormEvent<EventTarget>) => {
    e.preventDefault()
    submitPwReset()
      .then(results => {
        console.log(results)
        if (results && results.data && results.data.submitPasswordReset) {
          setMessage(
            'Check your email inbox, we have sent you a password reset link.'
          )
        }
      })
      .catch(err => console.log(err))
  }

  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        {loading && (
          <Styled.SignInSignUpLoading>
            <LoadingSpinner />
          </Styled.SignInSignUpLoading>
        )}

        <Styled.SignInSignUpText>
          Reset your password here, <br />
          <span>just type your email</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm
          onSubmit={handleSubmit}
          className={loading ? 'loading' : ''}
        >
          <Styled.SignInSignUpForm__Input>
            <input
              type="email"
              placeholder="Email address"
              value={emailVal}
              onChange={e => setEmailVal(e.target.value)}
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Send
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>

        {message && (
          <Styled.SignInSignUpMessage>
            <CheckIcon />
            <p>{message}</p>
          </Styled.SignInSignUpMessage>
        )}

        {error && (
          <Styled.SignInSignUpMessage>
            <ErrorIcon />
            <p>{error.message}</p>
          </Styled.SignInSignUpMessage>
        )}
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default ForgotPassword
