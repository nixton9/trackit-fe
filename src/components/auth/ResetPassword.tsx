import React, { useState } from 'react'
import { SignInSignUpHeader } from '../misc/SignInSignUpHeader'
import { LoadingSpinner } from '../misc/LoadingSpinner'
import { Styled } from '../../styles/SignInSignUp.styles'
import { ReactComponent as ErrorIcon } from '../../assets/icons/error.svg'
import { ReactComponent as CheckIcon } from '../../assets/icons/check.svg'
import { gql, useMutation } from '@apollo/client'
import { useParams } from 'react-router'

const RESET_PASSWORD = gql`
  mutation resetPassword(
    $email: String!
    $resetToken: String!
    $newPassword: String!
  ) {
    resetPassword(
      email: $email
      resetToken: $resetToken
      newPassword: $newPassword
    ) {
      id_user
    }
  }
`

type ResetPwData = {
  resetPassword: {
    id_user: string
  }
}

const ResetPassword: React.FC = () => {
  const [passwordVal, setPasswordVal] = useState('')
  const [message, setMessage] = useState('')

  const { email, token } = useParams()

  const [resetPw, { error, loading }] = useMutation<ResetPwData>(
    RESET_PASSWORD,
    {
      variables: {
        email: email,
        resetToken: token,
        newPassword: passwordVal
      }
    }
  )

  const handleSubmit = (e: React.FormEvent<EventTarget>) => {
    e.preventDefault()
    resetPw()
      .then(results => {
        if (results && results.data && results.data.resetPassword) {
          setMessage('Your password was updated successfully.')
        }
      })
      .catch(err => console.log(err))
  }
  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        {loading && (
          <Styled.SignInSignUpLoading>
            <LoadingSpinner />
          </Styled.SignInSignUpLoading>
        )}

        <Styled.SignInSignUpText>
          Set your new password,
          <br />
          <span>make it sticky this time</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm
          onSubmit={handleSubmit}
          className={loading ? 'loading' : ''}
        >
          <Styled.SignInSignUpForm__Input>
            <input
              type="password"
              placeholder="Password"
              value={passwordVal}
              onChange={e => setPasswordVal(e.target.value)}
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Update
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>

        {message && (
          <Styled.SignInSignUpMessage>
            <CheckIcon />
            <p>{message}</p>
          </Styled.SignInSignUpMessage>
        )}

        {error && (
          <Styled.SignInSignUpMessage>
            <ErrorIcon />
            <p>{error.message}</p>
          </Styled.SignInSignUpMessage>
        )}
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default ResetPassword
