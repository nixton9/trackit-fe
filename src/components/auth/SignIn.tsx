import React, { useState, useRef } from 'react'
import { SignInSignUpHeader } from '../misc/SignInSignUpHeader'
import { LoadingSpinner } from '../misc/LoadingSpinner'
import { Styled } from '../../styles/SignInSignUp.styles'
import { ReactComponent as EyeIcon } from '../../assets/icons/eye.svg'
import { ReactComponent as ErrorIcon } from '../../assets/icons/error.svg'
import { LOGIN } from '../../utils/queries'
import { User } from '../../utils/ModuleTypes'
import { useMutation } from '@apollo/client'
import { Link } from 'react-router-dom'

export type SignInSignUpProps = {
  setToken: (token: string) => void
  setUserInfo: (user: User) => void
  setNotToken: (not_token: string | null) => void
}

type SignInData = {
  login: {
    token: string
    user: {
      id_user: string
      name_user: string
      email_user: string
      image_user: string
      not_token: string
    }
  }
}

const SignIn: React.FC<SignInSignUpProps> = ({
  setToken,
  setUserInfo,
  setNotToken
}) => {
  const [emailVal, setEmailVal] = useState('')
  const [passwordVal, setPasswordVal] = useState('')

  const passwordEl = useRef<HTMLInputElement>(null)
  const iconEl = useRef<SVGSVGElement>(null)

  const [login, { error, loading }] = useMutation<SignInData>(LOGIN, {
    variables: { email: emailVal, password: passwordVal }
  })

  const togglePasswordVisibility = () => {
    const input = passwordEl.current
    const icon = iconEl.current
    if (input && input.type === 'password') {
      input.type = 'text'
      icon?.classList.add('selected')
    } else if (input) {
      input.type = 'password'
      icon?.classList.remove('selected')
    }
  }

  const handleLogin = (e: React.FormEvent<EventTarget>) => {
    e.preventDefault()
    login()
      .then(results => {
        if (results && results.data && results.data.login) {
          setToken(results.data.login.token)
          setUserInfo({
            id: results.data.login.user.id_user,
            name: results.data.login.user.name_user,
            email: results.data.login.user.email_user,
            image: results.data.login.user.image_user
          })
          setNotToken(results.data.login.user.image_user || null)
        }
      })
      .catch(err => console.log(err))
  }

  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        {loading && (
          <Styled.SignInSignUpLoading>
            <LoadingSpinner />
          </Styled.SignInSignUpLoading>
        )}

        <Styled.SignInSignUpText>
          Welcome back, <br />
          <span>sign in to continue</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm
          onSubmit={handleLogin}
          className={loading ? 'loading' : ''}
        >
          <Styled.SignInSignUpForm__Input>
            <input
              type="email"
              placeholder="Email address"
              value={emailVal}
              onChange={e => setEmailVal(e.target.value)}
              data-test-id="signin-email"
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <div className="input-wrapper">
              <input
                type="password"
                placeholder="Password"
                value={passwordVal}
                onChange={e => setPasswordVal(e.target.value)}
                ref={passwordEl}
                data-test-id="signin-pw"
              />
              <EyeIcon onClick={togglePasswordVisibility} ref={iconEl} />
            </div>
            <p>
              <Link to="/forgot">Forgot password?</Link>
            </p>
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Login
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>

        {error && (
          <Styled.SignInSignUpMessage>
            <ErrorIcon />
            <p>
              {error.message.includes('Unexpected token')
                ? "We're sorry but it seems there was a problem reaching the server"
                : error.message}
            </p>
          </Styled.SignInSignUpMessage>
        )}
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default SignIn
