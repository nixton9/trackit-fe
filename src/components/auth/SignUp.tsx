import React, { useState, useRef } from 'react'
import { SignInSignUpHeader } from '../misc/SignInSignUpHeader'
import { LoadingSpinner } from '../misc/LoadingSpinner'
import { Styled } from '../../styles/SignInSignUp.styles'
import { ReactComponent as EyeIcon } from '../../assets/icons/eye.svg'
import { ReactComponent as ErrorIcon } from '../../assets/icons/error.svg'
import { SignInSignUpProps } from './SignIn'
import { gql, useMutation } from '@apollo/client'
import { useHistory } from 'react-router-dom'

const REGISTER = gql`
  mutation Signup($name: String!, $email: String!, $password: String!) {
    signup(name: $name, email: $email, password: $password) {
      user {
        id_user
        name_user
        email_user
        image_user
      }
      token
    }
  }
`
type SignUpData = {
  signup: {
    token: string
    user: {
      id_user: string
      name_user: string
      email_user: string
      image_user: string
    }
  }
}

const SignUp: React.FC<SignInSignUpProps> = ({ setToken, setUserInfo }) => {
  const [nameVal, setNameVal] = useState('')
  const [emailVal, setEmailVal] = useState('')
  const [passwordVal, setPasswordVal] = useState('')
  const [formError, setFormError] = useState('')

  const passwordEl = useRef<HTMLInputElement>(null)
  const iconEl = useRef<SVGSVGElement>(null)

  const [register, { error, loading }] = useMutation<SignUpData>(REGISTER, {
    variables: { name: nameVal, email: emailVal, password: passwordVal }
  })

  const history = useHistory()

  const togglePasswordVisibility = () => {
    const input = passwordEl.current
    const icon = iconEl.current
    if (input && input.type === 'password') {
      input.type = 'text'
      icon?.classList.add('selected')
    } else if (input) {
      input.type = 'password'
      icon?.classList.remove('selected')
    }
  }

  const handleRegister = (e: React.FormEvent<EventTarget>) => {
    e.preventDefault()

    if (passwordVal.length < 6) {
      setFormError('Password needs to have at least 6 characters')
    } else {
      setFormError('')
      register()
        .then(results => {
          if (results && results.data && results.data.signup) {
            setToken(results.data.signup.token)
            setUserInfo({
              id: results.data.signup.user.id_user,
              name: results.data.signup.user.name_user,
              email: results.data.signup.user.email_user,
              image: results.data.signup.user.image_user
            })
            history.push('/')
          }
        })
        .catch(err => console.log(err))
    }
  }

  return (
    <>
      <SignInSignUpHeader />

      <Styled.SignInSignUpContainer>
        {loading && (
          <Styled.SignInSignUpLoading>
            <LoadingSpinner />
          </Styled.SignInSignUpLoading>
        )}

        <Styled.SignInSignUpText>
          Hey there, <br />
          <span>sign up to continue</span>
        </Styled.SignInSignUpText>

        <Styled.SignInSignUpForm
          onSubmit={handleRegister}
          className={loading ? 'loading' : ''}
        >
          <Styled.SignInSignUpForm__Input>
            <input
              type="text"
              placeholder="Name"
              value={nameVal}
              onChange={e => setNameVal(e.target.value)}
              required
              data-test-id="signup-name"
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <input
              type="email"
              placeholder="Email address"
              value={emailVal}
              onChange={e => setEmailVal(e.target.value)}
              required
              data-test-id="signup-email"
            />
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Input>
            <div className="input-wrapper">
              <input
                type="password"
                placeholder="Password"
                value={passwordVal}
                onChange={e => setPasswordVal(e.target.value)}
                ref={passwordEl}
                required
                data-test-id="signup-pw"
              />
              <EyeIcon onClick={togglePasswordVisibility} ref={iconEl} />
            </div>
          </Styled.SignInSignUpForm__Input>

          <Styled.SignInSignUpForm__Button>
            Sign Up
          </Styled.SignInSignUpForm__Button>
        </Styled.SignInSignUpForm>

        {error ? (
          <Styled.SignInSignUpMessage>
            <ErrorIcon />
            <p>{error.message}</p>
          </Styled.SignInSignUpMessage>
        ) : (
          formError && (
            <Styled.SignInSignUpMessage>
              <ErrorIcon />
              <p>{formError}</p>
            </Styled.SignInSignUpMessage>
          )
        )}
      </Styled.SignInSignUpContainer>
    </>
  )
}

export default SignUp
