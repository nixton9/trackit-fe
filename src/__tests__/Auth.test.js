import React from 'react'
import { render } from '@testing-library/react'
import SignIn from '../components/auth/SignIn'
import SignUp from '../components/auth/SignUp'
import ForgotPassword from '../components/auth/ForgotPassword'
import ResetPassword from '../components/auth/ResetPassword'
import { BrowserRouter } from 'react-router-dom'
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: '/',
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjYsImlhdCI6MTU5NDYyMzYxN30.eRuuyy-6EE6UkGFQI9UQmRklwPiyXgSJgmoZektpiG4`
  }
})
describe('SignIn', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SignIn />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the inputs', () => {
    const { getByPlaceholderText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SignIn />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByPlaceholderText('Email address')).toBeInTheDocument()
    expect(getByPlaceholderText('Password')).toBeInTheDocument()
  })
})

describe('SignUp', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the inputs', () => {
    const { getByPlaceholderText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByPlaceholderText('Name')).toBeInTheDocument()
    expect(getByPlaceholderText('Email address')).toBeInTheDocument()
    expect(getByPlaceholderText('Password')).toBeInTheDocument()
  })
})

describe('ForgotPassword', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <ForgotPassword />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has the email input', () => {
    const { getByPlaceholderText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <ForgotPassword />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByPlaceholderText('Email address')).toBeInTheDocument()
  })
})

describe('ResetPassword', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <ResetPassword />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has the password input', () => {
    const { getByPlaceholderText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <ResetPassword />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByPlaceholderText('Password')).toBeInTheDocument()
  })
})
