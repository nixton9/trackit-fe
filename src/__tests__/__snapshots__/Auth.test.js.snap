// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword matches snapshot 1`] = `
<div>
  <nav
    class="sc-AxjAm icODTr"
  >
    <h1
      class="sc-AxirZ bosuDb"
    >
      TrackIt
    </h1>
    <div
      class="sc-AxiKw cmYvTE"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Sign In
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </nav>
  <div
    class="sc-AxhCb ivLleu"
  >
    <h2
      class="sc-AxhUy JNaYu"
    >
      Reset your password here, 
      <br />
      <span>
        just type your email
      </span>
    </h2>
    <form
      class="sc-AxgMl iZsHBs"
    >
      <div
        class="sc-AxheI fnSDUS"
      >
        <input
          placeholder="Email address"
          type="email"
          value=""
        />
      </div>
      <button
        class="sc-Axmtr lgMnpa"
      >
        Send
      </button>
    </form>
    
  </div>
</div>
`;

exports[`ResetPassword matches snapshot 1`] = `
<div>
  <nav
    class="sc-AxjAm icODTr"
  >
    <h1
      class="sc-AxirZ bosuDb"
    >
      TrackIt
    </h1>
    <div
      class="sc-AxiKw cmYvTE"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Sign In
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </nav>
  <div
    class="sc-AxhCb ivLleu"
  >
    <h2
      class="sc-AxhUy JNaYu"
    >
      Set your new password,
      <br />
      <span>
        make it sticky this time
      </span>
    </h2>
    <form
      class="sc-AxgMl iZsHBs"
    >
      <div
        class="sc-AxheI fnSDUS"
      >
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        class="sc-Axmtr lgMnpa"
      >
        Update
      </button>
    </form>
    
  </div>
</div>
`;

exports[`SignIn matches snapshot 1`] = `
<div>
  <nav
    class="sc-AxjAm icODTr"
  >
    <h1
      class="sc-AxirZ bosuDb"
    >
      TrackIt
    </h1>
    <div
      class="sc-AxiKw cmYvTE"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Sign In
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </nav>
  <div
    class="sc-AxhCb ivLleu"
  >
    <h2
      class="sc-AxhUy JNaYu"
    >
      Welcome back, 
      <br />
      <span>
        sign in to continue
      </span>
    </h2>
    <form
      class="sc-AxgMl iZsHBs"
    >
      <div
        class="sc-AxheI fnSDUS"
      >
        <input
          placeholder="Email address"
          type="email"
          value=""
        />
      </div>
      <div
        class="sc-AxheI fnSDUS"
      >
        <div
          class="input-wrapper"
        >
          <input
            placeholder="Password"
            type="password"
            value=""
          />
          <svg>
            eye.svg
          </svg>
        </div>
        <p>
          <a
            href="/forgot"
          >
            Forgot password?
          </a>
        </p>
      </div>
      <button
        class="sc-Axmtr lgMnpa"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;

exports[`SignUp matches snapshot 1`] = `
<div>
  <nav
    class="sc-AxjAm icODTr"
  >
    <h1
      class="sc-AxirZ bosuDb"
    >
      TrackIt
    </h1>
    <div
      class="sc-AxiKw cmYvTE"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Sign In
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </nav>
  <div
    class="sc-AxhCb ivLleu"
  >
    <h2
      class="sc-AxhUy JNaYu"
    >
      Hey there, 
      <br />
      <span>
        sign up to continue
      </span>
    </h2>
    <form
      class="sc-AxgMl iZsHBs"
    >
      <div
        class="sc-AxheI fnSDUS"
      >
        <input
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-AxheI fnSDUS"
      >
        <input
          placeholder="Email address"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="sc-AxheI fnSDUS"
      >
        <div
          class="input-wrapper"
        >
          <input
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
          <svg>
            eye.svg
          </svg>
        </div>
      </div>
      <button
        class="sc-Axmtr lgMnpa"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
